src/popup/App.vue

```diff
@@ -591,54 +591,72 @@ export default {
       let that = this;
       that.loading = true;
 
-      let res = await fetch(
-        that.apiUrl + "/api/v2/alias/custom/new?hostname=" + that.hostName,
-        {
-          method: "POST",
-          body: JSON.stringify({
-            alias_prefix: that.aliasPrefix,
-            signed_suffix: that.signedSuffix,
-          }),
-          headers: {
-            "Content-Type": "application/json",
-            Authentication: this.apiKey,
-          },
-        }
-      );
-
-      let json = await res.json();
-      that.loading = false;
-      if (res.status == 201) {
-        that.newAlias = json.alias;
-        that.recomputeShowVoteScreen();
-      } else {
-        that.showError(json.error);
-      }
+      axios
+        .post(
+          that.apiUrl + "/api/v2/alias/custom/new?hostname=" + that.hostName,
+          { alias_prefix: that.aliasPrefix, signed_suffix: that.signedSuffix },
+          {
+            headers: { Authentication: this.apiKey },
+          }
+        )
+        .then((res) => {
+          if (res.status == 201) {
+            that.newAlias = res.data.alias;
+            that.recomputeShowVoteScreen();
+          } else {
+            that.showError(res.data.error);
+          }
+        })
+        .catch((err) => {
+          // rate limit reached
+          if (err.request.status == 429) {
+            that.showError(
+              "Rate limit exceeded - please wait 60s before creating new alias"
+            );
+          } else if (err.request.status == 409) {
+            that.showError("Alias already chosen, please select another one");
+          } else {
+            that.showError("Unknown error");
+          }
+        })
+        .then(() => {
```

Comment:
I think that we can use `.finally()` here

Reply:
`finally` seems to be finally 🙃  supported by `axios` https://github.com/axios/axios/issues/34. 

However, in its doc on https://github.com/axios/axios, it still recommends using `then` as a way to implement `finally`. To stay on the safe track, I think we can move to `finally` once it's officially recommended in axios doc.

![image](https://user-images.githubusercontent.com/3167780/85526422-48786780-b60a-11ea-9061-1f18e79d83aa.png)